### ISSUE TITLE
Set Undertow max Parameter

### ISSUE THREAD

**Issue by munendrasn** [decebals/pippo#359]
Using `pippo-undertow` in my project. I want to increase default `MAX_PARAMETER` limit.

Only way to achieve this is to extend `UndertowServer` and setting `serverOptions` in createServer.
Is there a better way to do this??

Probably like adding in `application.properties` and picking the value using `pippoSettings`.
Then we could set all serverOptions in `createServer()` itself

**Comment by decebals** [comment]
Your approach is good.
Is it a possibility to use `application.properties` to define options and use `setServerOption` (or `setSocketOption`, `setWorkerOption`) from `Undertow.Builder` class.
We can use a prefix in application.properties, like:
```
undertow.server.<option_name_1>=...
undertow.server.<option_name_2>=...
undertow.socket.<option_name_3>=...
undertow.socket.<option_name_4>=...
undertow.worker.<option_name_5>=...
undertow.worker.<option_name_6>=...
```

In `UndertowSettings` you can add `getServerOptions` (`getSocketOptions`, `getWorkerOptions`) method|(s) that get all settings with prefix `undertow.server` (`undertow.socket`, `undertow.worker`) using [PippoSettings.getSettingNames(String startingWith)](https://github.com/decebals/pippo/blob/master/pippo-core/src/main/java/ro/pippo/core/PippoSettings.java#L418), remove the prefix from each returned setting and use http://docs.jboss.org/xnio/3.0.4.GA/api/org/xnio/Option.html#fromString(java.lang.String,java.lang.ClassLoader) to convert a string option in an Option instance.

It's just a suggestion, I am not sure if it works.

**Comment by munendrasn** [comment]
I will try this approach and revert back to you with results

**Comment by munendrasn** [comment]
> In UndertowSettings you can add getServerOptions (getSocketOptions, getWorkerOptions) method|(s) that get all settings with prefix undertow.server (undertow.socket, undertow.worker) using PippoSettings.getSettingNames(String startingWith), remove the prefix from each returned setting and use http://docs.jboss.org/xnio/3.0.4.GA/api/org/xnio/Option.html#fromString(java.lang.String,java.lang.ClassLoader) to convert a string option in an Option instance.

Before trying out this approach, a small question. We would get `List<String>` from `PippoSettings.getSettings()` and convert them to `Options`. But we wont be knowing about the `option` type right??
Same would be true for Option value(both would be same)

For Example,
Consider this `application.properties` file
```
undertow.server.MAX_PARAMETER=2000
undertow.server.ENABLE_HTTP2=true
```
using PippoSettings.getSettingNames("undertow.server"), we would get both `MAX_PARAMETER` and `ENABLE_HTTP2`.
Now, how do we convert to `Option<Integer>` and `Option<Boolean>`??

Only way is to maintain a mapping between `settingsName` and `settingsType`
If i'm wrong please correct me

**Comment by decebals** [comment]
You have right. I don't know for moment how to detect the option type. I don't know if http://atetric.com/atetric/javadoc/org.jboss.xnio/xnio-api/3.3.6.Final/org/xnio/Options.html helps.

**Comment by decebals** [comment]
Another interesting class is [UndertowOptions](https://repository.jboss.org/nexus/content/unzip/unzip/io/undertow/undertow-core/1.0.14.Final/undertow-core-1.0.14.Final-javadoc.jar-unzip/io/undertow/UndertowOptions.html).

**Comment by munendrasn** [comment]
That would be good idea. But Is it possible to handle this pippo itself as `UndertowOptions` is in `undertow`??

**Comment by decebals** [comment]
> But Is it possible to handle this pippo itself as UndertowOptions is in undertow??

I don't understand above sentence. Could you add more details?

**Comment by decebals** [comment]
Did you find a solution?

**Comment by decebals** [comment]
Resolved by #384
